gpMgmt/bin/lib/gp_bash_functions.sh

```diff
@@ -193,13 +193,35 @@ WARN_MARK="<<<<<"
 # Functions
 #******************************************************************************
 
+#
+# Simplified version of _nl_normalize_codeset from glibc
+#https://sourceware.org/git/?p=glibc.git;a=blob;f=intl/l10nflist.c;h=078a450dfec21faf2d26dc5d0cb02158c1f23229;hb=1305edd42c44fee6f8660734d2dfa4911ec755d6#l294
+# Input parameter - string with locale defined as [language[_territory][.codeset][@modifier]]
+NORMALIZE_CODESET_IN_LOCALE () {
+	local language_and_territory=$(echo $1 | perl -ne 'print for /(^.+?(?=\.|@|$))/s')
+	local codeset=$(echo $1 | perl -ne 'print for /((?<=\.).+?(?=@|$))/s')
+	local modifier=$(echo $1 | perl -ne 'print for /((?<=@).+(?=$))/s' )
```

Comment:
The pattern `/((?<=@).+(?=$))/s` adds new line symbol (`0x0A`) to the end of matched substring

Reply:
As evidence:
```bash
echo 'en_US.UTF-8@123' | perl -ne 'print for /((?<=\.).+(?=@|$))/s' | hexdump -C
00000000  31 32 33 0a                                       |123.|
00000004
```
the new line is preserved from `echo` call, to remove it use `-n` option.

Reply:
Enclosing of end of line patter `$` into look-ahead assertion `(?=...)` looks redundant.

Reply:
If we add ? for pattern - end of line will not be included

```
echo 'en_US.UTF-8@123' | perl -ne 'print for /((?<=@).+(?=$))/s' | hexdump -C
00000000  31 32 33 0a                                       |123.|
00000004

echo 'en_US.UTF-8@123' | perl -ne 'print for /((?<=@).+?(?=$))/s' | hexdump -C
00000000  31 32 33                                          |123|
00000003

```

Reply:
But variant with add -n option for echo will look better

Reply:
> If we add ? for pattern - end of line will not be included

Yes, as another option. But IMO the lazy mode for pattern capture is redundant here.